packages/matchbox/src/components/Popover/Popover.js

```diff
@@ -1,159 +1,153 @@
+import { createPropTypes } from '@styled-system/prop-types';
+import { Box } from '../Box';
 import PopoverOverlay from './PopoverOverlay';
 import PopoverContent from './PopoverContent';
 import { onKey } from '../../helpers/keyEvents';
+import useWindowEvent from '../../hooks/useWindowEvent';
+import { deprecate } from '../../helpers/propTypes';
 
-import styles from './Popover.module.scss';
-
-class Popover extends Component {
-  static displayName = 'Popover';
-
-  static propTypes = {
-    /**
-     * A React component to will trigger the popover
-     * Click event is handled for you if this component is uncontrolled.
-     */
-    trigger: PropTypes.node,
-    /**
-      * Adds a padding to the Popover
-      */
-    sectioned: PropTypes.bool,
-    /**
-      * Opens the popover.
-      * By default, open state is handled automatically. Passing this value in will turn this into a controlled component.
-      */
-    open: PropTypes.bool,
-    left: PropTypes.bool,
-    right: PropTypes.bool,
-    top: PropTypes.bool,
-    bottom: PropTypes.bool,
-    /**
-      * Callback function that is called when clicking outside the popover, or hitting escape.
-      */
-    onClose: PropTypes.func,
-    /**
-      * Popover Content
-      */
-    children: PropTypes.node,
-    /**
-     * Element ID for the portal that will house Popover. Appends to body if not provided.
-     */
-    portalId: PropTypes.string
-  };
-
-  static defaultProps = {
-    right: true,
-    bottom: true
-  }
+function Popover(props) {
+  const { as, id, open: controlledOpen, onClose, children, trigger, wrapper, ...rest } = props;
+  const [open, setOpen] = React.useState(null);
+  const popoverRef = React.useRef();
+  const activatorRef = React.useRef();
 
-  state = {
-    open: null
-  }
+  const shouldBeOpen = controlledOpen || open;
+  const Wrapper = as || wrapper || 'span';
 
-  componentDidMount() {
-    const { open: controlledOpen } = this.props;
+  useWindowEvent('click', handleOutsideClick);
+  useWindowEvent('keydown', handleEsc);
 
+  // Mount hook – sets initial close state if uncontrolled
+  React.useEffect(() => {
     // This component becomes "controlled" if the prop 'open' is given a boolean value
     if (controlledOpen === undefined) {
-      this.setState({ open: false });
+      setOpen(false);
     }
-  }
+  }, []);
 
-  uncontrolledToggle = () => {
-    this.setState({ open: !this.state.open });
+  // Toggles uncontrolled open state
+  function handleUncontrolledToggle() {
+    setOpen(!open);
   }
 
-  handleOutsideClick = (e) => {
-    const { open: controlledOpen, onClose } = this.props;
-    const isOutside = this.popover && !this.popover.contains(e.target) && this.activator && !this.activator.contains(e.target);
+  // Toggles uncontrolled popovers, and calls `onClose` for controlled popovers
+  function handleOutsideClick(e) {
+    const isOutside =
+      !popoverRef.current.contains(e.target) && !activatorRef.current.contains(e.target);
```

Comment:
Since the listener is on the window, these unfortunately need the guards to check if the refs exist.
<img width="1391" alt="Screen Shot 2020-04-06 at 1 22 21 PM" src="https://user-images.githubusercontent.com/1335605/78586634-b5e5da80-7809-11ea-9d83-d848dcfbcebe.png">